gum/valgrind.h

```diff
@@ -691,31 +691,59 @@ typedef
+volatile unsigned long long int  _zzq_result;                 \
+_zzq_args[0] = (unsigned long long int)(_zzq_request);        \
+_zzq_args[1] = (unsigned long long int)(_zzq_arg1);           \
+_zzq_args[2] = (unsigned long long int)(_zzq_arg2);           \
+_zzq_args[3] = (unsigned long long int)(_zzq_arg3);           \
+_zzq_args[4] = (unsigned long long int)(_zzq_arg4);           \
+_zzq_args[5] = (unsigned long long int)(_zzq_arg5);           \
+__asm__ volatile("mov x3, %1\n\t" /*default*/                 \
+                "mov x4, %2\n\t" /*ptr*/                     \
+                __SPECIAL_INSTRUCTION_PREAMBLE               \
+                /* X3 = client_request ( X4 ) */             \
+                "orr x10, x10, x10\n\t"                      \
+                "mov %0, x3"     /*result*/                  \
+                : "=r" (_zzq_result)                         \
+                : "r" ((unsigned long long int)_zzq_default),\
+                  "r" (&_zzq_args[0])                        \
+                : "cc","memory", "x3", "x4");                \
+_zzq_result;                                                  \
+})
+#else
+# define VALGRIND_DO_CLIENT_REQUEST_EXPR(                                   \
+   _zzq_default, _zzq_request,                                         \
+   _zzq_arg1, _zzq_arg2, _zzq_arg3, _zzq_arg4, _zzq_arg5)              \
+                                                                       \
+__extension__                                                             \
+({volatile unsigned long long int  _zzq_args[6];                          \
+volatile unsigned long long int  _zzq_result;                           \
+_zzq_args[0] = (unsigned long long int)(_zzq_request);                  \
+_zzq_args[1] = (unsigned long long int)(unsigned long int)(_zzq_arg1);  \
+_zzq_args[2] = (unsigned long long int)(unsigned long int)(_zzq_arg2);  \
+_zzq_args[3] = (unsigned long long int)(unsigned long int)(_zzq_arg3);  \
+_zzq_args[4] = (unsigned long long int)(unsigned long int)(_zzq_arg4);  \
+_zzq_args[5] = (unsigned long long int)(unsigned long int)(_zzq_arg5);  \
+__asm__ volatile("mov x3, %1\n\t" /*default*/                           \
+                "mov x4, %2\n\t" /*ptr*/                               \
+                __SPECIAL_INSTRUCTION_PREAMBLE                         \
+                /* X3 = client_request ( X4 ) */                       \
+                "orr x10, x10, x10\n\t"                                \
+                "mov %0, x3"     /*result*/                            \
+                : "=r" (_zzq_result)                                   \
+                : "r" ((unsigned long long int)_zzq_default),          \
+                  "r" (&_zzq_args[0])                                  \
+                : "cc","memory", "x3", "x4");                          \
+_zzq_result;                                                            \
+})
```

Comment:
For the added code, let's use the exact same indentation as the original, so it's easier to diff.